ui-admin/src/study/participants/survey/PrintFormModal.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react'
+import { Survey as SurveyComponent } from 'survey-react-ui'
+
+import {
+  surveyJSModelFromForm,
+  makeSurveyJsData,
+  waitForImages,
+  Survey,
+  ConsentForm,
+  configureModelForPrint
+} from '@juniper/ui-core'
+
+import { Answer } from 'api/api'
+import Modal from 'react-bootstrap/Modal'
+import { Link } from 'react-router-dom'
+
+
+type DownloadFormViewProps = {
+    answers: Answer[],
+    resumeData?: string,
+    survey: Survey | ConsentForm
+}
+
+
+/** renders the form in a fullscreen modal, and pops up the print dialog */
+const PrintFormModal = ({ survey, answers, resumeData }: DownloadFormViewProps) => {
+  const surveyJsData = makeSurveyJsData(resumeData, answers, undefined)
+  const surveyJsModel = surveyJSModelFromForm(survey)
+  surveyJsModel.data = surveyJsData.data
+  configureModelForPrint(surveyJsModel)
+  useEffect(() => {
+    waitForImages().then(() => { window.print() })
+  }, [surveyJsModel])
```

Comment:
```suggestion
  useEffect(() => {
    waitForImages().then(() => { printForm() })
  }, [surveyJsModel])

  function printForm() {
    const initialPageTitle = document.title
    //The filename defaults to whatever the page title is, so this temporarily sets it to the form name
    document.title = `${survey.name}.pdf`
    window.print()
    document.title = initialPageTitle
  }
```

Not sure if this is too hacky, but this would be a decent UX improvement IMO. Right now the downloaded form is titled "Juniper.pdf" by default.

Reply:
great idea, done!